src/google-maps.ts

```diff
@@ -275,27 +301,60 @@ export class GoogleMaps {
      * @param geocoder any
      *
      */
-    geocodeAddress(address: string, geocoder: any) {
-        this._mapPromise.then(() => {
-            geocoder.geocode({'address': address}, (results: any, status: string) => {
-                if (status !== (<any>window).google.maps.GeocoderStatus.OK) {
-                    return;
-                }
+    geocodeAddress(address: string) {
+        this.geocode(address).then(firstResult => {
+            this.setCenter(firstResult.geometry.location);
+            this.createMarker({
+                map: this.map,
+                position: firstResult.geometry.location
+            }).then((createdMarker: any) => {
+                this._locationByAddressMarkers.push(createdMarker);
+                this.eventAggregator.publish(LOCATIONADDED, Object.assign(createdMarker, { placeId: firstResult.place_id }));
+            });
+        });
+    }
 
-                let firstResultLocation = results[0].geometry.location;
+    /**
+     * Geocodes Address and returns the coordinates once the google map has been properly initialized
+     *
+     * @param address string
+     *
+     */
+    addressToMarker(address: string): Promise<LatLongMarker> {
+        return this.geocode(address).then(firstResults => {
+            return {
+                latitude: firstResults.geometry.location.lat(),
+                longitude: firstResults.geometry.location.lng()
+            }
+        });
+    }
 
-                this.setCenter(firstResultLocation);
-                this.createMarker({
-                    map: this.map,
-                    position: firstResultLocation
-                }).then((createdMarker: any) => {
-                    this._locationByAddressMarkers.push(createdMarker);
-                    this.eventAggregator.publish(LOCATIONADDED, Object.assign(createdMarker, {placeId: results[0].place_id}));
+    /**
+     * Geocodes Address and returns the firstresults object after google maps has initialized
+     *
+     * @param address string
+     *
+     */
+    private geocode(address: string): Promise<any> {
+        return this._mapPromise.then(() => {
+            return new Promise((resolve, reject) => {
+                this.geocoder.geocode({ 'address': address }, (results: any, status: string) => {
+                    if (status !== (<any>window).google.maps.GeocoderStatus.OK) {
+                        reject(new Error(``));
```

Comment:
Does `GeocoderStatus` return a proper error reason. If Yes use this information in `new Error()`